File: app/src/main/java/com/greenaddress/jade/HttpRequestHandler.java
package com.greenaddress.jade;

import com.fasterxml.jackson.databind.JsonNode;

import java.io.IOException;

// HttpRequestHandler is used for network calls during pinserver handshake
// useful on TOR enabled sessions
public interface HttpRequestHandler{
    JsonNode httpRequest(final JsonNode details) throws IOException;
}

File: common/src/commonMain/kotlin/com/blockstream/common/interfaces/HttpRequestHandler.kt
package com.blockstream.common.interfaces

import kotlinx.serialization.json.JsonElement

// HttpRequestHandler is used for network calls during pinserver handshake
// useful on TOR enabled sessions
interface HttpRequestHandler {
    fun prepareHttpRequest()

    fun httpRequest(details: JsonElement): JsonElement

    fun httpRequest(
        method: String,
        urls: List<String>?,
        data: String?,
        accept: String?,
        certs: List<String>?
    ): JsonElement
}